tools/walletextension/subscriptions/subscriptions.go

```diff
@@ -0,0 +1,88 @@
+package subscriptions
+
+import (
+	"errors"
+	"fmt"
+	gethlog "github.com/ethereum/go-ethereum/log"
+	"github.com/obscuronet/go-obscuro/go/common"
+	"github.com/obscuronet/go-obscuro/go/common/log"
+	"github.com/obscuronet/go-obscuro/go/rpc"
+	wecommon "github.com/obscuronet/go-obscuro/tools/walletextension/common"
+	"github.com/obscuronet/go-obscuro/tools/walletextension/userconn"
+)
+
+type SubscriptionManager struct {
+	logger gethlog.Logger
+}
+
+func New(logger gethlog.Logger) *SubscriptionManager {
+	return &SubscriptionManager{
+		logger: logger,
+	}
+}
+
+func (s *SubscriptionManager) HandleNewSubscriptions(clients []rpc.Client, req *wecommon.RPCRequest, resp *interface{}, userConn userconn.UserConn) error {
+	if len(req.Params) == 0 {
+		return fmt.Errorf("could not subscribe as no subscription namespace was provided")
+	}
+
+	if !userConn.SupportsSubscriptions() {
+		return errors.New("userConn doesn't support subscriptions")
+	}
+
```

Comment:
This method shouldn't receive connections that don't support subscriptions.

Reply:
True, removed if statement